client.cc

```diff
@@ -11,22 +11,48 @@
 
 static int have_response = 0;
 
+#ifndef COAP_CLIENT_URI
+#define COAP_CLIENT_URI "coap://coap.me/hello"
```

Comment:
Try changing the URI to "coap://coap.me/hello/xyz"

Reply:
The server `coap.me` does not have a resource defined for `/hello/xyz`, and so will return a "`4.04 not found`".